third_party/txt/src/txt/paragraph.cc

```diff
@@ -1123,7 +1143,18 @@ std::vector<Paragraph::TextBox> Paragraph::GetRectsForRange(size_t start,
 
   std::vector<Paragraph::TextBox> boxes;
   for (const auto& kv : line_boxes) {
-    boxes.insert(boxes.end(), kv.second.begin(), kv.second.end());
+    if (rect_style & RectStyle::kTight) {
+      // Ignore line max height and width and generate tight bounds.
+      boxes.insert(boxes.end(), kv.second.begin(), kv.second.end());
+    } else {
+      // Set each box to the max height of each line to ensure continuity.
+      size_t line = kv.first;
+      for (const Paragraph::TextBox& box : kv.second) {
```

Comment:
You can calculate the minimum top and maximum bottom values for the rects on the line here instead of maintaining the min_top and max_bottom vectors above